images/router/haproxy/conf/haproxy-config.template

```diff
@@ -155,8 +175,13 @@ backend be_edge_http_{{$cfgIdx}}
   mode http
   balance leastconn
   timeout check 5000ms
+  {{ if (eq $cfg.TLSTermination "") }}
+  cookie OPENSHIFT_{{$cfgIdx}}_SERVERID insert indirect nocache httponly
+  {{ else }}
+  cookie OPENSHIFT_EDGE_{{$cfgIdx}}_SERVERID insert indirect nocache httponly secure
+  {{ end }}
```

Comment:
Hmm, do we want to make the sticky sessions behavior the default? Wouldn't this mean that now all client connections would end up being fixed to the same backend instance? This does have implications if there's a mix of short and long running sessions - we could very well end up with an unbalanced haproxy setup - example one backend instance gets "favored" over all others. 
Also, most applications at scale understand the need to remove the dependency on sticky sessions or use some sort of external scalable session store. Yeah, there'll be a few that require sticky sessions but IMHO, rather than make that the default, it make more sense to have that be the exception.
One option would be to have a field on a ```route``` that says it needs ```session-affinity`` and based on that generate this config.